ledger/ledgercore/statedelta.go

```diff
@@ -191,19 +193,26 @@ type AccountDeltas struct {
 	assetResourcesCache map[AccountAsset]int
 }
 
-// MakeStateDelta creates a new instance of StateDelta.
+// MakeStateDelta creates a new instance of StateDelta
 // hint is amount of transactions for evaluation, 2 * hint is for sender and receiver balance records.
 // This does not play well for AssetConfig and ApplicationCall transactions on scale
-func MakeStateDelta(hdr *bookkeeping.BlockHeader, prevTimestamp int64, hint int, stateProofNext basics.Round) StateDelta {
-	return StateDelta{
-		Accts: MakeAccountDeltas(hint),
-		Txids: make(map[transactions.Txid]IncludedTransactions, hint),
-		// asset or application creation are considered as rare events so do not pre-allocate space for them
-		Hdr:                      hdr,
-		StateProofNext:           stateProofNext,
-		PrevTimestamp:            prevTimestamp,
-		initialTransactionsCount: hint,
-	}
+func MakeStateDelta(hdr *bookkeeping.BlockHeader, prevTimestamp int64, hint int, stateProofNext basics.Round) (sd StateDelta) {
+	sd = StateDelta{}
+	sd.PopulateStateDelta(hdr, prevTimestamp, hint, stateProofNext)
+	return
+}
+
+// PopulateStateDelta populates an existing StateDelta struct.
+// Used as a helper for MakeStateDelta as well as for re-using already allocated structs from sync.Pool
+// Make sure to .Reset the struct before re-use
+func (sd *StateDelta) PopulateStateDelta(hdr *bookkeeping.BlockHeader, prevTimestamp int64, hint int, stateProofNext basics.Round) {
+	sd.Accts = MakeAccountDeltas(hint)
+	sd.Txids = make(map[transactions.Txid]IncludedTransactions, hint)
```

Comment:
I suppose there is always a map in `sd.Txids` already.  Would it be cheaper to clear it than to allocate a new one?